### PR TITLE
feat(core): package direction

### PR DESCRIPTION
**Pull request by ramzzo**
close https://github.com/oku-ui/primitives/issues/243

### CODE REVIEW

**Comment by productdevbook** [comment]
After checking this, let's ask to write a test if everything is correct @dammy001

File: packages/components/direction/src/Direction.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+const DirectionContextSymbol = Symbol('DirectionContext')
+
+/* -------------------------------------------------------------------------------------------------
+ * Direction
+ * ----------------------------------------------------------------------------------------------- */
+
+const DirectionProvider = defineComponent({
+  name: 'DirectionProvider',
+  props: {
+    dir: { type: String as () => Direction, required: true },
```
Comment (dammy001) [inline packages/components/direction/src/Direction.ts:13]:
```suggestion
    dir: { type: String as PropType<Direction>, required: true },
```

File: packages/components/direction/src/Direction.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { defineComponent, inject, provide } from 'vue'
```
Comment (dammy001) [inline packages/components/direction/src/Direction.ts:1]:
```suggestion
import { defineComponent, inject, provide, type PropType } from 'vue'
```

**Comment by dammy001** [comment]
@ramzzo can you add a few tests for this?

**Comment by ramzzo** [comment]
> @ramzzo can you add a few tests for this?

I have not written a test before, but I would like to try it.

**Review by productdevbook — APPROVED** [review]

**Comment by productdevbook** [comment]
:heart:

**Pull request merged**